src/graphql/template.rs

```diff
@@ -448,4 +453,107 @@ mod tests {
         let res = schema.execute(r#"{templateList{totalCount}}"#).await;
         assert_eq!(res.data.to_string(), r#"{templateList: {totalCount: 0}}"#);
     }
+
+    #[tokio::test]
+    async fn test_structured_template() {
+        let schema = TestSchema::new().await;
+        let res = schema.execute(r#"{templateList{totalCount}}"#).await;
+        assert_eq!(res.data.to_string(), r#"{templateList: {totalCount: 0}}"#);
+
+        let res = schema
+            .execute(
+                r#"mutation {
+                    insertTemplate(structured: {
+                        name: "t1",
+                        description: "test",
+                        algorithm: "OPTICS",
+                        eps: 0.5,
+                        format: "json",
+                        timeIntervals: [1, 2, 3],
+                        numbersOfTopN: [1, 2, 3]
+                    })
+                }"#,
+            )
+            .await;
+        assert_eq!(res.data.to_string(), r#"{insertTemplate: "t1"}"#);
+
+        let res = schema.execute(r#"{templateList{totalCount}}"#).await;
+        assert_eq!(res.data.to_string(), r#"{templateList: {totalCount: 1}}"#);
+
+        let res = schema
+            .execute(
+                r#"{
+                templateList {
+                    edges {
+                        node {
+                            ... on StructuredClusteringTemplate {
+                                name
+                            }
+                        }
+                    }
+                totalCount
+            }
+        }"#,
+            )
+            .await;
+        assert_eq!(
+            res.data.to_string(),
+            r#"{templateList: {edges: [{node: {name: "t1"}}], totalCount: 1}}"#
+        );
+
+        let res = schema
+            .execute(
+                r#"mutation {
+                updateTemplate(oldStructured: {
+                    name: "t1",
+                    description: "test",
+                    algorithm: "OPTICS",
+                    eps: 0.5,
+                    format: "json",
+                    timeIntervals: [1, 2, 3],
+                    numbersOfTopN: [1, 2, 3]
+                },
+                newStructured: {
+                    name: "t1",
+                    description: "test",
+                    algorithm: "OPTICS",
+                    eps: 0.5,
+                    format: "json",
+                    timeIntervals: [1, 2, 4],
+                    numbersOfTopN: [1, 2, 4]
+                })
+            }"#,
+            )
+            .await;
+        assert_eq!(res.data.to_string(), r#"{updateTemplate: true}"#);
+
+        let res = schema
+            .execute(
+                r#"{
+                templateList {
+                    edges {
+                        node {
+                            ... on StructuredClusteringTemplate {
+                                algorithm
```

Comment:
If you import the `timeIntervals` field, it is expected to be included in the test.

Reply:
@marioCluml I think this comment is related to this question https://github.com/aicers/review-web/pull/270#issuecomment-2301141916

Reply:
The test with `timeIntervals` for `UnstructuredClusteringTemplate` is added. There are two test functions in `mod tests` : `test_structured_template` and `test_unstructured_templates`.

Reply:
Does it have to be on one test for CodeCov to recognize `timeIntervals` is included in the test? I believe `timeIntervals` can be tested but not model.rs.

Reply:
I told you because I thought it would be necessary to test this part.
![스크린샷 2024-08-29 오전 9 31 18](https://github.com/user-attachments/assets/5aa74e6e-af34-4eb2-9609-71d0be02358b)

Reply:
Ah you mean for `StructuredClusteringTemplate`

Reply:
Done